src/main/java/org/ice4j/ice/ConnectivityCheckClient.java

```diff
@@ -739,8 +739,11 @@ else if (!parentAgent.isControlling()
         }
 
         // Selected pairs get their consent freshness confirmed.
-        if (checkedPair == checkedPair.getParentComponent().getSelectedPair())
+        // XXX Should we also confirm consent freshness for non-selected pairs?
+        if (checkedPair.equals(checkedPair.getParentComponent().getSelectedPair()))
```

Comment:
nice catch.  your comment sounds right...if we expect to be able to fall back, do we need to be carrying this out for the non-selected pairs as well?